clang/runtime/dpct-rt/include/dpct/device.hpp

```diff
@@ -785,6 +785,33 @@ has_capability_or_fail(const sycl::device &dev,
     break;
   }
 }
+
+/// For USM, if \p queue is the default queue, waits all the kernel tasks
+/// in all the queues of current device completed, then stores the event of
+/// the queue specified by \p queue at the time of this call into the memory
+/// pointed by \p event_ptr. If \p queue is not the default queue, just stores
+/// the event of the queue specified by \p queue at the time of this call into
+/// the memory pointed by \p event_ptr.
+/// For usmnone, waits all the kernel tasks
+/// in all the queues of current device completed, then stores the event of the
+/// queue specified by \p queue at the time of this call into memory pointed by
+/// \p event_ptr.
+inline void sycl_event_record(dpct::event_ptr event_ptr,
+                              sycl::queue *queue = &get_default_queue()) {
+#ifdef DPCT_USM_LEVEL_NONE
+  dpct::get_current_device().queues_wait_and_throw();
```

Comment:
why need sync here?

Reply:
As out-of-ordered queue is used for USM-none,  sync is used to make sure all the all the kernel tasks completed before storing queue event.